src/test/unit_tests/test_nc_health.js

```diff
@@ -601,19 +591,107 @@ mocha.describe('nsfs nc health', function() {
             const account_invalid = { name: 'account_invalid', uid: 999, gid: 999, allow_bucket_creation: true };
             await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.ADD, { config_root, ...account_invalid });
-            Health.get_service_state.restore();
-            Health.get_endpoint_response.restore();
             Health.all_account_details = true;
             Health.all_bucket_details = false;
-            const get_service_state = sinon.stub(Health, "get_service_state");
-            get_service_state.onFirstCall().returns(Promise.resolve({ service_status: 'active', pid: 1000 }))
-                .onSecondCall().returns(Promise.resolve({ service_status: 'active', pid: 2000 }));
-            const get_endpoint_response = sinon.stub(Health, "get_endpoint_response");
-            get_endpoint_response.onFirstCall().returns(Promise.resolve({response: {response_code: 'RUNNING', total_fork_count: 0}}));
+            set_mock_functions(Health, {
+                get_service_state: get_service_state_mock_default_response,
+                get_endpoint_response: get_endpoint_response_mock_default_response,
+                get_system_config_file: get_system_config_mock_default_response
+            });
             const health_status = await Health.nc_nsfs_health();
             assert.strictEqual(health_status.checks.accounts_status.valid_accounts.length, 1);
             assert.strictEqual(health_status.checks.accounts_status.invalid_accounts.length, 1);
             await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.DELETE, { config_root, name: account_invalid.name });
         });
+
+        mocha.it('Health all condition - failed config directory upgrade status', async function() {
+            valid_system_json.config_directory = {
+                'config_dir_version': config_fs.config_dir_version,
+                'upgrade_package_version': pkg.version,
+                'phase': CONFIG_DIR_LOCKED,
+                upgrade_history: {
+                    successful_upgrades: [],
+                    last_failure: { error: 'mock error'}
+                }
+            };
+            await Health.config_fs.create_system_config_file(JSON.stringify(valid_system_json));
+            const health_status = await Health.nc_nsfs_health();
+            await fs_utils.file_delete(Health.config_fs.system_json_path);
+            assert_config_dir_status(health_status, valid_system_json.config_directory);
+        });
+
+        mocha.it('Health all condition - valid ongoing config directory upgrade', async function() {
+            valid_system_json.config_directory = {
+                config_dir_version: config_fs.config_dir_version,
+                upgrade_package_version: pkg.version,
+                phase: CONFIG_DIR_LOCKED,
+                upgrade_history: {
+                    successful_upgrades: [],
+                },
+                in_progress_upgrade: { from_version: 'x', to_version: 'y'}
```

Comment:
Maybe add a real version details?
```suggestion
                in_progress_upgrade: { from_version: '5.17.8', to_version: '5.18.0'}
```